pkg/log/checkpoint/checkpoint.go

```diff
@@ -47,3 +125,17 @@ func hostName() string {
 
 	return hostname
 }
+
+func logCheckPointError(logger log.Logger, err error) {
+	logger.Log("checkpoint error", err)
+}
+
+func parseUrl(addr string) (*url.URL, error) {
+	return url.Parse(fmt.Sprintf("https://%s", strings.Split(addr, ":")[0]))
+}
+
+func logMap(logger log.Logger, entry map[string]interface{}) {
```

Comment:
I don't like what this does to the log output. The map should get passed as a map -- flattening it here causes different data to end up on the same keys. I don't think it's needed